client/relay-chain-rpc-interface/src/tokio_platform.rs

```diff
@@ -0,0 +1,329 @@
+// Copyright 2023 Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+use core::time::Duration;
+use futures::{channel::mpsc, prelude::*, task::Poll};
+use smoldot::libp2p::{multiaddr::ProtocolRef, websocket, Multiaddr};
+use smoldot_light::platform::{
+	ConnectError, PlatformConnection, PlatformSubstreamDirection, ReadBuffer,
+};
+use std::{
+	collections::VecDeque,
+	io::IoSlice,
+	net::{IpAddr, SocketAddr},
+	pin::Pin,
+	sync::Arc,
+};
+use tokio::net::TcpStream;
+
+use tokio_util::compat::TokioAsyncReadCompatExt;
+
+/// Platform internal representation of the connection stream.
+pub struct TokioStream {
+	shared: Arc<StreamShared>,
+	read_data_rx: Arc<parking_lot::Mutex<stream::Peekable<mpsc::Receiver<Vec<u8>>>>>,
+	read_buffer: Option<Vec<u8>>,
+}
+
+struct StreamShared {
+	guarded: parking_lot::Mutex<StreamSharedGuarded>,
+	write_queue_pushed: event_listener::Event,
+}
+
+struct StreamSharedGuarded {
+	write_queue: VecDeque<u8>,
+}
+
+/// Platform implementation for tokio
+/// This implementation is a conversion of the implementation for async-std:
+/// https://github.com/smol-dot/smoldot/blob/54d88891b1da202b4bf612a150df7b4dbfa03a55/light-base/src/platform/async_std.rs#L40
+pub struct TokioPlatform;
+
+impl smoldot_light::platform::Platform for TokioPlatform {
```

Comment:
As a heads up I'm going to change the trait a bit definition: https://github.com/smol-dot/smoldot/pull/315
The implementation in the smoldot repo is also much cleaner now.

Reply:
Fair 👍, I will adjust here accordingly. Thanks for notifying!